### ISSUE TITLE
Escaping issues with OAuth header

### ISSUE THREAD

**Issue by mmd-osm** [osmlab/osm-auth#64]
Follow up for https://github.com/zerebubuth/openstreetmap-cgimap/issues/146 and https://github.com/zerebubuth/openstreetmap-cgimap/pull/147

Apologies, this issue is a bit confusing, as I identified the root cause in the meantime, and updated the issue a couple of times.

Here's the original issue:

_I'm using osm-auth for unit testing, i.e. sending some HTTP POST request to the OSM API with OAuth support. This all works fine. However, we've identified an issue with the following HTTP POST payload:_

```
<dummy id='1'/>
```

_For some reason, parts of this string also make it into the OAuth Authentication header, and worse, there appears to be some issue with the way the apostrophe (`'`) character is being transmitted. If I'm using `"` instead of `'` for the id attribute, everything works fine._

Original xhr call triggering the issue: https://github.com/mmd-osm/osm-auth/blob/6b69fcf23d5c6e33014afa053a5ea08a3a53983c/index.html#L93-L96

I'm suspecting that there was some issue in the way I use this library. Looking at the iD source, here's how the upload call is implemented there. 

```javascript
oauth.xhr({
                method: 'POST',
                path: '/api/0.6/changeset/' + changesetID + '/upload',
                options: { header: { 'Content-Type': 'text/xml' } },
                content: JXON.stringify(changeset.osmChangeJXON(changes))
```

Once I added the `options` field to my own call, the issue seems to be gone now. I'm not sure, if I also would need the JXON.stringify. Updated call: https://github.com/mmd-osm/osm-auth/blob/debug_client/index.html#L93-L97

So in conclusion, I think it would be pretty good to have some working example code for an OSM API upload call as part of the documentation in this library.
